stdlib/sql/vertica.go

```diff
@@ -0,0 +1,170 @@
+package sql
+
+import (
+	"database/sql"
+	"fmt"
+	"time"
+
+	"github.com/influxdata/flux"
+	"github.com/influxdata/flux/codes"
+	"github.com/influxdata/flux/execute"
+	"github.com/influxdata/flux/internal/errors"
+	"github.com/influxdata/flux/values"
+)
+
+type VerticaRowReader struct {
+	Cursor      *sql.Rows
+	columns     []interface{}
+	columnTypes []flux.ColType
+	columnNames []string
+}
+
+func (m *VerticaRowReader) Next() bool {
+	next := m.Cursor.Next()
+	if next {
+		columnNames, err := m.Cursor.Columns()
+		if err != nil {
+			return false
+		}
+		m.columns = make([]interface{}, len(columnNames))
+		columnPointers := make([]interface{}, len(columnNames))
+		for i := 0; i < len(columnNames); i++ {
+			columnPointers[i] = &m.columns[i]
+		}
+		if err := m.Cursor.Scan(columnPointers...); err != nil {
+			return false
+		}
+	}
+	return next
+}
+
+func (m *VerticaRowReader) GetNextRow() ([]values.Value, error) {
+	row := make([]values.Value, len(m.columns))
+	for i, col := range m.columns {
+		switch col := col.(type) {
+		case bool, int, uint:
+			row[i] = values.New(int64(col.(int)))
+		case int64, uint64, float64:
+			row[i] = values.New(col)
+		case string:
+			row[i] = values.New(col)
+		case time.Time:
+			row[i] = values.NewTime(values.ConvertTime(col))
+		case []uint8:
+			fmt.Print("hack:", m.columnTypes[i])
+			// Hack for MySQL, might need to work with charset?
+			// Can't do boolean with MySQL - stores BOOLEANs as TINYINTs (0 or 1)
+			// No way to distinguish if intended int or bool
+			switch m.columnTypes[i] {
+			case flux.TInt:
+				newInt, err := UInt8ToInt64(col)
+				if err != nil {
+					return nil, err
+				}
+				row[i] = values.NewInt(newInt)
+			case flux.TFloat:
+				newFloat, err := UInt8ToFloat(col)
+				if err != nil {
+					return nil, err
+				}
+				row[i] = values.NewFloat(newFloat)
+			// This works, but you can also just just add the DSN parameter parseTime=true (see line 136)
+			case flux.TTime:
+				t, err := time.Parse(layout, string(col))
+				if err != nil {
+					fmt.Print(err)
+				}
+				row[i] = values.NewTime(values.ConvertTime(t))
+			default:
+				row[i] = values.NewString(string(col))
+			}
```

Comment:
My understanding of this code block is that, if we get a slice of bytes, we can't necessarily tell what type it is, so we check the flux column type and attempt to convert to that. I follow the logic there, and the code seems good, but I have some questions I'd like to answer before this merges.

From what I can tell, the code block (including the comment referencing MySQL), was originally added to the `mysql` package, then it got copied and pasted over to the `postgres` package, and I would guess that the same thing happened here.

If this 'hack' was only introduced for the sake of compatibility with MySQL, do we really need it here? Looking at some of the earlier comments on this PR, it's suggested that this code is unreachable when interacting with veritca. I'm not familiar with vertica... can we confirm that's the case? Is it impossible for vertica to have a column of bytes?

If it turns out this code is useful outside of MySQL, then let's keep it and just remove the misleading comments. But if we can confirm that it's unreachable, I think it would be best to simplify here. Maybe just convert it to a string and call it done?